### DISCUSSION TITLE
An easier way to skip the next alarm

### DISCUSSION THREAD

**Discussion by MoonlitMonitor** [WrichikBasu/ShakeAlarmClock#76] [Feature requests]
Currently, if I want to skip the next alarm on a recurring alarm schedule (e.g. once per day), I have to open Shake Alarm Clock, switch off the alarm, wait until the alarm would have otherwise played, then switch the alarm back on. A problem with this approach is that the user might forget to switch the alarm back on for future usage.

A more efficient way to do this would be to display a persistent notification notifying you about your upcoming alarm, with the option to dismiss it underneath. That way, if the user has a daily alarm for example, but they want to skip the upcoming alarm, they can just press "Dismiss" and they do not need to perform the extra step of switching the alarm back on.

A real-life example of where this feature might be useful is if you have a daily alarm, say at 12 PM, to water a plant. Assume that sometimes you have to leave your house at 11:45 AM, so you water the plant right before you leave the house. However the next day you will be at home at 12 PM to water the plant. In this case, you may want to dismiss today's alarm prior to 12 PM so that it is does not ring at 12 PM today, but you would still want the alarm to ring tomorrow at 12 PM. Simply dismissing the alarm will have this effect, and you will not be required to turn the alarm back on so that it rings normally, unlike now.

**Comment by WrichikBasu** [comment]
Thank you for using the app! The idea is nice. Not sure how long the notification will stay due to restrictions placed by the Android system, but definitely worth a try. I'll implement this.

**Reply by MoonlitMonitor** [reply]
Thank you. I am running GrapheneOS and certain notifications persist until I swipe them away. Also, in the default Clock app on GrapheneOS, when I set an alarm, it seems like the notification only shows up within about 2 hours of when the alarm is intended to sound. Having said that, if there is no restriction on persistence, then it might be a good idea to let the user decide whether they want:

- the notification to show permanently
- the notification to show a few hours before the alarm is intended to sound (maybe even a custom duration?)
- the notification to be hidden
